src/psyclone/tests/psyir/conftest.py

```diff
@@ -0,0 +1,51 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2019-2020, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author A. R. Porter, STFC Daresbury Lab
+
+
+''' Module which performs pytest set-up specific to the PSyIR tests. '''
+
+from __future__ import absolute_import
+import pytest
+from psyclone.psyir.nodes import Node
+from psyclone.psyir.symbols import DataSymbol, DataType
+
+@pytest.fixture(scope="function")
+def disable_declaration_check(request, monkeypatch):
+    ''' By default a Reference checks that it has a corresponding entry in
```

Comment:
I'm not sure I like this idea. Is it really that bad to require symbol table entries? Perhaps an issue should be created rather than "fixing" it this way?

Reply:
I currently use it on some 50 tests, some of which are parametrised. We (I) can of course correct all of these tests if need be. Why are you not sure you like it?

Reply:
As discussed I think it is a bit of a hack, but I agree that it involves a lot of changes to fix so please create an issue and we can defer for the moment.

Reply:
Created #754.

Reply:
Great. I've added a new rule to the reviewing guidelines on the wiki that says that if a test using this fixture is modified then it must be changed to not require the fixture.